addons/lunch/models/lunch_product.py

```diff
@@ -20,44 +21,51 @@ class LunchProduct(models.Model):
     _name = 'lunch.product'
     _description = 'Lunch Product'
 
-    name = fields.Char('Product', required=True)
-    category_id = fields.Many2one('lunch.product.category', 'Product Category', required=True)
+    name = fields.Char('Name', required=True)
+    category_id = fields.Many2one('lunch.product.category', 'Product Category')
     description = fields.Text('Description')
-    price = fields.Float('Price', digits=dp.get_precision('Account'))
-    supplier = fields.Many2one('res.partner', 'Vendor')
+    price = fields.Float('Price', digits=dp.get_precision('Account'), required=True)
+    supplier = fields.Many2one('lunch.supplier', 'Vendor', required=True)
     active = fields.Boolean(default=True)
-    available = fields.Boolean(compute='_get_available_product', search='_search_available_products')
+    is_topping = fields.Boolean("This product is an extra garniture")
+
+    # image: all image fields are base64 encoded and PIL-supported
+    image = fields.Binary(
+        "Image", attachment=True,
+        help="This field holds the image used as image for the product, limited to 1024x1024px.")
+    image_medium = fields.Binary(
+        "Medium-sized image", attachment=True,
+        help="Medium-sized image of the product. It is automatically "
+             "resized as a 128x128px image, with aspect ratio preserved, "
+             "only when the image exceeds one of those sizes. Use this field in form views or some kanban views.")
+    image_small = fields.Binary(
+        "Small-sized image", attachment=True,
+        help="Small-sized image of the product. It is automatically "
+             "resized as a 64x64px image, with aspect ratio preserved. "
+             "Use this field anywhere a small image is required.")
+
+    is_favorite = fields.Boolean('Has Already Been Ordered', compute='_compute_is_favorite')
```

Comment:
Why don't call it "already_ordered" ? This is not like the user explicitly add the product to its favorite list, as it is deduced from its old orders.